lib/pure/collections/deques.nim

```diff
@@ -62,6 +62,12 @@ type
     data: seq[T]
     head, tail, count, mask: int
 
+proc initImpl[T](deq: var Deque[T], initialSize: int = 4) =
+  if deq.mask == 0:
+    assert isPowerOfTwo(initialSize)
```

Comment:
Should it be a `doAssert` :grey_question: :thinking:

Reply:
I think not. This was already there, I just moved the implementation to a separate proc.